### ISSUE TITLE
A way to override form submission email sending

### ISSUE THREAD

**Issue by duncanmcclean** [statamic/ideas#680]
It would be nice if there was a way developers could override the saving of form submissions. Similar how they could in v2 with the `Form.submission.creating` event.

## Use Case

I'm porting over a site from v2 to v3 which has quite a bit of custom logic around forms. 

One of those bits of logic is where we change the email form submissions get sent to. We have a form where the user selects a 'Branch'. Then in our v2 listener, we get the entry for that branch, grab it's email field and we replace `{{ branch_email }}` with the branch's email address (if the branch is one of the configured email addresses).

I had a look but there's currently only a single event for submissions: `SubmissionCreated` which isn't very helpful (I'm happy to include support for Submissions in statamic/cms#3379 btw if you think that would work)

## Workaround

Right now, I've done a bit of a hacky workaround. Extended the form tag to use a custom controller which extends Statamic's `FormController` that then dispatches a custom version of the `SendEmails` job which I then have a three line check in to replace `{{ branch_email }}` with what it should be. 😅

**Comment by ryanmitchell** [comment]
So we do this a lot in forms, and you can do this as things stand....

listen for the `FormSubmitted` event, then do something like:

```
public function handle(FormSubmitted &$event)
{

    $event->submission->form->email([
    // new email array
    ]); 

}
```

Not sue the & in front of the param is entirely needed but old habits die hard.

The values you pass to form->email() are used for the lifecycle of the object so your notification recipient will be whatever you pass.

**Comment by duncanmcclean** [comment]
@ryanmitchell Oh, that's nice. I'll take a look when I have some time - would certainly be better than what we're doing at present (however, I think I ended up doing even more customisations 😆 )

**Comment by ryanmitchell** [comment]
@duncanmcclean Yeah its proven useful since I discovered it. We use postmark for sending emails with template vars, so we hit their api with the right variables and then blank out the email array in the form to stop statamic sending emails. Works really well.
